cpu/sam3: add ADC implementation

haukepetersen opened the pull request.

aabadie commented on boards/arduino-due/include/arduino_pinmap.h:
```diff
@@ -113,6 +114,24 @@ extern "C" {
+ * @name    Mapping of Ardunino analog pins to RIOT ADC lines
+ * @{
+ */
+#define ARDUINO_AN0             ADC_LINE(7)
```
I would use `ARDUINO_A0` to follow Arduino names.

aabadie commented:
Just tested it and saw inconsistencies with Arduino pin names:
- board A7 is controlled by ARDUINO_A0
- board A0 is controlled by ARDUINO_A7 => A0 to A7 are inverted
- board A8 is controlled by ARDUINO_A10
- board A9 is controlled by ARDUINO_A11
- board A10 is controlled by ARDUINO_A12
- board A11 is controlled by ARDUINO_A13

aabadie commented:
ping @haukepetersen, any chance to fix the aforementioned issues ?

haukepetersen commented:
Changed the Arduino pin names to `ARDUINO_Ax` and squashed right in.

> Just tested it and saw inconsistencies with Arduino pin names

nope, no inconsistencies, all as it should be. See e.g. https://dl.dropboxusercontent.com/u/50261731/Wiki%20Work/Due-pinout_tinyG2_0.2.svg - arduino pin A0 (green box) is connected to SAMs analog line AD7 (white box) and so on... Just tested the mapping again and found it working as expected...

aabadie commented:
> arduino pin A0 (green box) is connected to SAMs analog line AD7 (white box)

Indeed, then the behavior is as expected.

aabadie commented:
and go!
